doc/developers/contributing.rst

```diff
@@ -751,8 +751,12 @@ In general have the following in mind:
 * When editing reStructuredText (``.rst``) files, try to keep line length under
   80 characters when possible (exceptions include links and tables).
 
-* Before submitting you pull request check if your modifications have introduced
-  new sphinx warnings and try to fix them.
+* Before submitting you pull request:
+
+    * check if your modifications have introduced new sphinx warnings and try
+      to fix them;
+    * check if you have removed or modified anchors for internal links and
+      revert them if needed.
```

Comment:
I wouldn't place it under the "before submitting your pull request"  but rather as a single entry: "Do not modify anchors for internal links as this would break existing URLs."

BTW is there a more common term than "anchor"? I'm not sure most users would understand what this refers to

Reply:
Not more common, but the sphinx vocabulary says "label": I should probably use "label" then.